services/node_stake_update.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "timestamp.proto";
+
+/**
+ * Updates the staking info at the end of staking period to indicate new staking period has started.
+ */
+message NodeStakeUpdateTransactionBody {
+  /**
+   * Time and date of the end of the staking period that is ending
+   */
+  Timestamp end_of_staking_period = 1;
+
+  /**
+   * Number of rounds during the staking period that is ending
+   */
+  uint32 num_staking_rounds = 2;
+
+  /**
+   * Staking info of each node at the beginning of the new staking period
+   */
+  repeated StakingNodeInfo staking_node_info = 3;
+}
+
+/**
+ * Staking info for each node for the staking period that is ending
+ */
+message StakingNodeInfo {
+
+  /**
+   * Node id of this node
+   */
+  uint64 node_id = 1;
+
+  /**
+   * Consensus weight for this node, at the end of current staking period.
+   */
+  uint64 stake = 2;
+
+  /**
+   * Total of balance of all accounts staked to this node that have declineReward set to false, at the end of staking period.
+   */
+  uint64 stake_rewarded_start = 3;
```

Comment:
I believe the HIP incorrectly refers to this as `stake_rewarded_start`. I believe it should just be `stake_rewarded` since it is equivalent to `stake` minus the not rewarded stake. It's different from the other similarly named `long stakedRewardedStart (value ≥ 0) stakedRewarded as of the start of the current staking period`.

```suggestion
  int64 stake_rewarded = 3;
```

Reply:
yes Agreed, `stake_rewarded_start` will send the starting stake of previous day , Is that your understanding as well ?

Reply:
Right